### DISCUSSION TITLE
关于SNN的MACs和FLOPs计算

### DISCUSSION THREAD

**Discussion by Maninnlack** [fangwei123456/SpikingFlow#211] [Q&A] [answered]
您好，最近在搭建SNN模型时想要计算SNN模型的MACs和FLOPs，来表示网络的复杂度以及进一步计算网络能耗。但是现在目前github上已有的计算MACs以及FLOPs方法都只能计算torch里面自带的module，所以想来请教一下这方面的计算方法，谢谢！

**Comment by fangwei123456** [comment, marked as answer]
多数和ANN相同，除了SNN特有的SOP。SOP是用脉冲神经元层的脉冲发放总次数（在测试集上统计），与下一层的FLOP相乘得到。

**Reply by Maninnlack** [reply]
您好！非常不好意思，我想问一下关于spikingjelly有关于统计脉冲神经网络SOP的函数吗？我在文档里面也没有找到关于统计脉冲发放总次数的内容，关于怎么计算SOP我实在是没有头绪。

**Reply by fangwei123456** [reply]
没有，需要借助统计flop的工具实现。统计发放率可以用新版的监视器实现：https://spikingjelly.readthedocs.io/zh_CN/fuse_ms/activation_based/monitor.html

监视器是独立的模块，可以把代码复制出来直接给老版本用

**Reply by Maninnlack** [reply]
多谢！这就去试试看！

**Reply by smile-cpu** [reply]
博主解决了嘛？

**Reply by Cmangsang** [reply]
您好！请问博主解决了吗
